pkg/providers/github.go

```diff
@@ -115,12 +114,34 @@ func newGitHub(u *url.URL) (Provider, error) {
 	}
 
 	token := os.Getenv("GITHUB_AUTH_TOKEN")
+
+	// GHES client
+	gbu := os.Getenv("GHES_BASE_URL")
+	guu := os.Getenv("GHES_UPLOAD_URL")
+	gau := os.Getenv("GHES_AUTH_TOKEN")
+
 	var tc *http.Client
-	if token != "" {
+
+	if len(gbu) > 0 && len(guu) > 0 && len(gau) > 0 {
+		tc = oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: token},
+		))
+	} else if token != "" {
 		tc = oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(
 			&oauth2.Token{AccessToken: token},
 		))
 	}
```

Comment:
Why not combine these conditions into one?
```suggestion

	if len(gbu) > 0 && len(guu) > 0 && len(gau) > 0 || token != "" {
		tc = oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(
			&oauth2.Token{AccessToken: token},
		))
	}
```

Reply:
because I made a mistake and I missed setting the correct token in the `valid` if statement clause :). Fixing